esp/platform/espp.cpp

```diff
@@ -360,8 +360,14 @@ void initializeMetrics(CEspConfig* config)
     //
     // Initialize metrics
     Owned<IPropertyTree> pMetricsTree = config->queryConfigPTree()->getPropTree("metrics");
```

Comment:
This is fine as it stands, but I would prefer it coded so that the component config is the default.  The reason is to make it consistent when the code is added to other components (and make it easier to common up).

```
    Owned<IPropertyTree> pMetricsTree = queryComponentConfig().getPropTree("metrics");
    if (pMetricsTree == nullptr)
    {
        pMetricsTree.setown(config->queryConfigPTree()->getPropTree("metrics"));
        queryComponentConfig().setPropTree("metrics", LINK(pMetricsTree));
    }

```

Reply:
Actually, don't bother with this change... leave it as you have it.